_internal/types.ts

```diff
@@ -255,12 +265,22 @@ export type RevalidateCallback = <K extends RevalidateEvent>(
   type: K
 ) => RevalidateCallbackReturnType[K]
 
-export type StateUpdateCallback<Data = any, Error = any> = (
-  state: State<Data, Error>
-) => void
-
-export interface Cache<Data = any, Error = any> {
-  get(key: Key): State<Data, Error> | undefined
-  set(key: Key, value: State<Data, Error>): void
+export interface Cache<Data = any> {
+  get(key: Key): CacheValue<Data> | undefined
+  set(key: Key, value: Data): void
   delete(key: Key): void
 }
+
+export interface CacheValue<Data = any, Error = any> {
+  data?: Data
+  error?: Error
+  isValidating?: boolean
+  isLoading?: boolean
+}
```

Comment:
`CacheValue` is identical to `State` (see https://github.com/vercel/swr/pull/1961/files#r872242431)

https://github.com/vercel/swr/blob/9a0328fffdeb8ca0eaf42bc88f318f7e48a3b245/_internal/types.ts#L177-L182

Reply:
These two types should be merged